### PR TITLE
Privileged containers should inherit host devices

### PR DESCRIPTION
**Pull request by baude**
When running a privileged container, it should inherit the same
devices the host has.

Signed-off-by: baude <bbaude@redhat.com>

### CODE REVIEW

File: test/e2e/privileged_test.go [hunk 1]
```diff
@@ -5,6 +5,7 @@ import (
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
+	//"strings"
```
Comment (mheon) [inline test/e2e/privileged_test.go]: Can you remove this?

File: test/e2e/privileged_test.go [hunk 2]
```diff
@@ -73,4 +74,17 @@ var _ = Describe("Podman privileged container tests", func() {
 		Expect(capAmp[1]).To(Equal(capEff[1]))
 	})
 
+	It("podman non-privileged should have very few devices", func() {
+		session := podmanTest.Podman([]string{"run", "busybox", "ls", "-l", "/dev"})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+		Expect(len(session.OutputToStringArray())).To(Equal(18))
```
Comment (mheon) [inline test/e2e/privileged_test.go]: Can we safely guarantee this is 18, or should we say under 20?

File: cmd/podman/spec.go [hunk 3]
```diff
@@ -685,3 +696,30 @@ func (c *createConfig) CreatePortBindings() ([]ocicni.PortMapping, error) {
 	}
 	return portBindings, nil
 }
+
+//AddPrivilegedDevices iterates through host devices and adds them to the spec
```
Comment (mheon) [inline cmd/podman/spec.go]: Nit: can you adjust the comment to say it adds all host devices to the spec?

File: cmd/podman/spec.go [hunk 4]
```diff
@@ -241,10 +246,16 @@ func createConfigToOCISpec(config *createConfig) (*spec.Spec, error) {
 	}
 
 	// Devices
-	for _, device := range config.Devices {
-		if err := addDevice(&g, device); err != nil {
+	if config.Privileged {
```
Comment (mheon) [inline cmd/podman/spec.go]: Can you add a comment on this saying there's no point in adding user-provided devices if we're adding all devices?

File: test/e2e/privileged_test.go (hunk 1 above)
Comment (baude) [reply test/e2e/privileged_test.go]: fixed

File: test/e2e/privileged_test.go (hunk 2 above)
Comment (baude) [reply test/e2e/privileged_test.go]: I want to see if 18 is firm.  Lets leave it there for now and see what happens.

File: cmd/podman/spec.go (hunk 3 above)
Comment (baude) [reply cmd/podman/spec.go]: sure.

File: cmd/podman/spec.go (hunk 4 above)
Comment (baude) [reply cmd/podman/spec.go]: sure

**Comment by mheon** [comment]
LGTM

**Comment by TomSweeneyRedHat** [comment]
LGTM and all happy green buttons.

**Comment by mheon** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**